File: Semestrul 3/MAP/lab 2/src/Containers/StackQueue.java
package Containers;

import Tasks.Task;

import java.util.ArrayList;
import java.util.List;
// TODO 5.3
public abstract class StackQueue implements Container {
    protected List<Task> tasks = new ArrayList<Task>();

    @Override
    public void add(Task task) {
        tasks.add(task);
    }

    @Override
    public int size() {
        return tasks.size();
    }

    @Override
    public boolean isEmpty() {
        return tasks.isEmpty();
    }
}

File: Semestrul 3/MAP/lab 10/containers/StackQueue.cs
namespace lab10;

public abstract class StackQueue : IContainer
{
    protected List<Task> Tasks = new List<Task>();

    public void Add(Task task)
    {
        Tasks.Add(task);
    }

    public int Size()
    {
        return Tasks.Count;
    }

    public bool IsEmpty()
    {
        return Tasks.Count == 0;
    }

    public abstract Task Remove();
}
